crates/dslab-core/src/state.rs

```diff
@@ -74,9 +76,53 @@ impl SimulationState {
         }
     }
 
+    pub fn add_ordered_event<T>(&mut self, data: T, src: Id, dest: Id, delay: f64) -> EventId
+    where
+        T: EventData,
+    {
+        if !self.can_add_ordered_event(delay) {
+            panic!("Event order is broken! Ordered events should be added in non-decreasing order of their time.");
+        }
+        let last_time = self.ordered_events.back().map_or(f64::MIN, |x| x.time);
+        let event_id = self.event_count;
+        let event = Event {
+            id: event_id,
+            time: last_time.max(self.clock + delay),
```

Comment:
Комментарий чтобы помнить

```suggestion
            // max is used to enforce time order despite of floating-point errors
            time: last_time.max(self.clock + delay),
```

Reply:
Добавил